src/Stream/MemoryStream.tpp

```diff
@@ -370,8 +329,103 @@ namespace Stream
       OutputMemoryStream(typename Allocator::size_type initial_size,
         const AllocatorInitializer& allocator_initializer)
       /*throw (eh::Exception)*/
-      : Holder(initial_size, allocator_initializer), Stream(this->buffer())
+      : Holder(initial_size, allocator_initializer)
+      , bad_(false)
+    {
+    }
+
+    template <typename Elem, typename Traits, typename Allocator,
+      typename AllocatorInitializer, const size_t SIZE>
+    void
+    OutputMemoryStream<Elem, Traits, Allocator, AllocatorInitializer, SIZE>::
+      append(Elem ch) /*throw (eh::Exception)*/
+    {
+      if (bad()) {
+        return;
+      }
+      auto* buffer = this->buffer();
+      if (buffer->ptr() == buffer->end() && !buffer->extend()) {
+        bad_ = true;
+        return;
+      }
+      *buffer->ptr() = ch;
+      buffer->incptr();
+    }
+
+    template <typename Elem, typename Traits, typename Allocator,
+      typename AllocatorInitializer, const size_t SIZE>
+    void
+    OutputMemoryStream<Elem, Traits, Allocator, AllocatorInitializer, SIZE>::
+      append(const Elem* str) /*throw (eh::Exception)*/
+    {
+      while (*str) {
```

Comment:
It is bad implementation - you make much unrequired checks for each char in string. Implement batch processing (buffer size check + memcpy)

Reply:
it was a temporary stub for me. now properly implemented as suggested.